Content.Server/Xenoarchaeology/XenoArtifacts/ArtifactComponent.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts;
+
+[RegisterComponent]
+public class ArtifactComponent : Component
+{
+    public override string Name => "Artifact";
+
+    /// <summary>
+    ///     Should artifact pick a random trigger on startup?
+    /// </summary>
+    [DataField("randomTrigger")]
+    public bool RandomTrigger = true;
+
+    /// <summary>
+    ///     List of all possible triggers activations.
+    ///     Should be same as components names.
+    /// </summary>
+    [DataField("possibleTriggers")]
+    public string[] PossibleTriggers = {
+        "ArtifactInteractionTrigger",
+        "ArtifactGasTrigger"
+    };
+
+    /// <summary>
+    ///     Cooldown time between artifact activations (in seconds).
+    /// </summary>
+    [DataField("timer")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public double CooldownTime = 10;
```

Comment:
some notion of 'max triggers' would be nice so you can't just get infinite of something

Reply:
Do you mean max activations count or max attached triggers components? If max activations - they should be infinite by design and only effects control their activation restrictions (check spawn effect that has max spawned items limit).

Reply:
ah ok then